src/BoundaryConditions/flux_calculations.jl

```diff
@@ -0,0 +1,70 @@
+# General
+"""
+    get_boundary_flux(model, ::NoFluxCondition, _...)
+
+Construct zero boundary fluxes for a scalar or 2-element vector(x,y)
+"""
+
+function get_boundary_flux(model, ::NoFluxCondition, var::Fields.Field, Ym, Ya)
+    FT = eltype(Ym)
+    flux = Geometry.Cartesian3Vector(FT(0))
+end
+
+# function get_boundary_flux(
+#     model,
+#     ::NoFluxCondition,
+#     ::Geometry.Cartesian12Vector,
+#     Ym,
+#     Ya,
+# )
+#     FT = eltype(Ym)
+#     flux = Geometry.Cartesian12Vector(FT(0), FT(0))
+# end
+
+# Momentum-specific
+"""
+    get_boundary_flux(model, ::DragLawCondition, _...)
+
+Construct non-zero boundary fluxes using the bulk formula and constant or Mohnin-Obukhov-based drag coefficient, Cd
+"""
+
+function get_boundary_flux(model, bc::DragLawCondition, uv, Ym, Ya)
+    FT = eltype(Ym)
+    coefficients = eltype(bc.coefficients) == FT ? bc.coefficients :
+        bc.coefficients(Ym, Ya)
+
+    uv_1 = Operators.getidx(Ym.uv, Operators.Interior(), 1)
+    u_wind = LinearAlgebra.norm(uv_1)
+
+    flux = Geometry.Cartesian3Vector(coefficients.Cd * u_wind) ⊗ uv_1
+end
+
+
+# Potential temperature density-specific
+"""
+    get_boundary_flux(model, ::DragLawCondition, _...)
+
+    Construct non-zero boundary fluxes using the bulk formula and constant or Mohnin-Obukhov-based heat transfer coefficient, Ch
+"""
+
+function get_boundary_flux(
+    model,
+    bc::BulkFormulaCondition,
+    ρθ::Fields.Field,
+    Ym,
+    Ya,
+)
+    FT = eltype(Ym)
+    coefficients = eltype(bc.coefficients) == FT ? bc.coefficients :
+        bc.coefficients(Ym, Ya)
+    T_sfc = bc.T_sfc
+
+    ρ_1 = Operators.getidx(Ym.ρ, Operators.Interior(), 1)
```

Comment:
similar - this is hard coded to apply only at the bottom of atmos, but here that's OK b/c of the type of flux.